x-pack/solutions/security/plugins/security_solution/server/lib/entity_analytics/privilege_monitoring/users/csv_upload.ts

```diff
@@ -46,18 +65,85 @@ export const createPrivilegedUsersCsvService = (dataClient: PrivilegeMonitoringD
       failed: 0,
       successful: 0,
     };
+
+    let runningUserCount = currentUserCount.count;
+    let batchNumber = 0;
+
     for await (const batch of batches) {
-      const usrs = await queryExistingUsers(esClient, index)(batch);
-      const upserted = await bulkUpsertBatch(esClient, index, options)(usrs);
-      results = accumulateUpsertResults(
-        { users: [], errors: [], failed: 0, successful: 0 },
-        upserted
+      batchNumber++;
```

Comment:
Would you be able to see if there is a way of containing as much of the upload limit logic in a separate function as possible? I worry this function is getting too complicated and has a lot of responsibility.

couold we potentially have a function which returns the batch with as many users as we can insert, alongside a list of records which couldn't be upserted due to the limit? 

I am also not sure about `runningCount` should we not check the count on each batch? E.g what if an index sync occurs while we are uploading